pydatastructs/linear_data_structures/tests/test_algorithms.py

```diff
@@ -31,3 +31,26 @@ def test_merge_sort_parallel():
                     709, 910]
     merge_sort_parallel(arr, 5, start=2, end=5)
     assert arr._data == expected_arr
+
+def test_brick_sort():
```

Comment:
May be we can use the test case in `test_merge_sort` for all the sorting algorithms. Something like this is also done with `minimum_spanning_tree`. Refer, https://github.com/codezonediitj/pydatastructs/blob/6605b9761a920632a69e34547765a2c2b78cbb13/pydatastructs/graphs/tests/test_algorithms.py#L154-L179

Reply:
ok so one common Array is sorted by multiple sorting algorithms, is that what you mean?

Reply:
Yes.

Reply:
shall I implement it in this PR itself or should I make another one?

Reply:
In this PR itself.

Reply:
Oh I get ur point, delete leaves None in the place where the value was originally, I might have to change many things in the code. Thanks it was very valuable input :)

Reply:
Probably, just `None`s have to hr ignored while comparing.

Reply:
One question. if [1,2,3,4,None,None,5,6] is the array and I sort from 3 to 6 should the None be in its original position or should it move to the end of the array.

Reply:
> Probably, just `None`s have to be ignored while comparing. `None`s can cause problems in many places. While porting to the code to C++, `NULL` can cause problems that are currently caused by `None`.

cant we overload the operators?

Reply:
another solution is to change delete to remove the element and reconfigure the array(thats how list is doing it)

Reply:
No, that shouldn't be done. The difference between `list` and DODA is the time complexity of deletion. `list` does the deletion in O(n) time but `DODA` does the same in amortized O(1) time. `None` is used as flag value in `DODA` that the element is deleted from a particular case.

Reply:
understood

Reply:
> One question. if [1,2,3,4,None,None,5,6] is the array and I sort from 3 to 6 should the None be in its original position or should it move to the end of the array.

and how should this be done?

Reply:
The following is what `merge_sort_parallel` does,
```python
>>> from pydatastructs import merge_sort_parallel
>>> from pydatastructs import DynamicOneDimensionalArray
>>> DODA = DynamicOneDimensionalArray
>>> arr = DODA(int, 0)
>>> arr.append(1)
>>> arr.append(2)
>>> arr.append(3)
>>> arr.append(4)
>>> arr.append(9)
>>> arr.append(10)
>>> arr.append(5)
>>> arr.append(6)
>>> arr._data
[1, 2, 3, 4, 9, 10, 5, 6, None, None, None, None, None, None, None]
>>> arr.delete(4)
>>> arr.delete(5)
>>> arr._data
[1, 2, 3, 4, None, None, 5, 6, None, None, None, None, None, None, None]
>>> merge_sort_parallel(arr, 1, start=2, end=7)
>>> arr._data
[1, 2, 3, 4, 5, 6, None, None, None, None, None, None, None]
```
`brick_sort` can do the same.

Reply:
merge_sort_parallel(arr, 1, start=2, end=5)
>>> arr._data
[1, 2, 3, 4, 5, 6, None, None, None, None, None, None, None]

as I was sorting from just 2 to 5 the array should have been [3,4,None,None,5] but the None values are moved to the end, so is that what you want to do or would it be better if the elements which are not included in the sort retain their original positions?

Reply:
Well, 2 to 5 includes `None`s, so change their positions should be expected.

Reply:
but that changes the positions of other elements as well, I can do it if u want but idk if users should expect other indexes to change if they sort a small segment.

Reply:
lets raise this issue some other time for some other PR, I will handle None's as you have done in merge sort

Reply:
I am looking into this issue for now. If it's fixed with minimal changes then we can go for it.

Reply:
Resolving.